languages/c/reference/README.md

```diff
@@ -0,0 +1,100 @@
+
+## Specialities (Concepts)
+
+- [Incomplete Types](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Incomplete-Types)
+- [Volatile](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Type-Qualifiers)
+- [Storage Class Specifiers](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Storage-Class-Specifiers)
+- [Renaming Types](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Renaming-Types)
+- [Bit Fields](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Bit-Fields)
+- [Labels](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Labels)
+- [The switch Statement](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#The-switch-Statement)
+- [Blocks](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Blocks)
+- [The Null Statement](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#The-Null-Statement)
+- [The goto Statement](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#The-goto-Statement)
+- [The typedef Statement](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#The-typedef-Statement)
+- [Variable Length Parameter Lists](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Variable-Length-Parameter-Lists)
+- [Calling Functions Through Function Pointers](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Calling-Functions-Through-Function-Pointers)
+- [Static Functions](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Static-Functions)
+- [Program Structure](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Program-Structure)
+- [Scope](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Scope)
+- [Dynamic memory management](https://en.cppreference.com/w/c/memory)
+- [Date and time utilities](https://en.cppreference.com/w/c/chrono)
+- [Strings library](https://en.cppreference.com/w/c/string)
+- [Algorithms](https://en.cppreference.com/w/c/algorithm)
+- [File input/output](https://en.cppreference.com/w/c/io)
+- [Undefined behaviour](https://en.wikipedia.org/wiki/Undefined_behavior)
+- [Macros](https://gcc.gnu.org/onlinedocs/cpp/Macros.html#Macros)
+- [The definition of `main()`](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#The-main-Function)
+- [C toolchain](https://progbook.org/prog2.html)
+- [Includes](https://gcc.gnu.org/onlinedocs/cpp/Include-Syntax.html)
+
+## Specialties (types)
+
+- [Complex Number Types](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Complex-Number-Types)
+- [Enumerations](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Enumerations)
+- [Unions](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Unions)
+
+## Specialities (patterns)
+
+- [Opaque pointers](https://en.wikipedia.org/wiki/Opaque_pointer)
+- [Callback](https://en.wikipedia.org/wiki/Callback_(computer_programming))
+
+## Concepts
```

Comment:
This is the same header as the `## Concepts` header at the beginning of the file. Maybe renamed this to "General Concepts" or something similar? You could also consider making this header (and it siblings) `###` headers (h3), as they seem to be categories in the more general `## Concepts` section.